.devcontainer/devcontainer.json

```
@@ -0,0 +1,7 @@
+{
+  "image": "mcr.microsoft.com/devcontainers/universal:2",
+  "features": {
+    },
+  "postCreateCommand": "python3 -m venv .venv & source .venv/bin/activate & pip3 install --upgrade pip & pip3 install -r requirements.txt",
```

So... there's a couple of issues here:
1. Since this is a container you don't strictly need to use virtual environment. Given that every second counts (devcontainers are pay per use) I would remove them.
2. You used `&` instead of `&&`. With `&` the process starts in the background therefore the `postCreateCommand` got stuck and I wasn't able to cleanly close the devcontainer.

```suggestion
  "postCreateCommand": "pip3 install --upgrade pip && pip3 install -r requirements.txt",
```